### PR TITLE
[NP-1885] add country regions

### PR DESCRIPTION
**Pull request by jinnnn9**
addresses: https://nanopay.atlassian.net/browse/NP-1885

### CODE REVIEW

File: src/foam/nanos/auth/Address.js [hunk 1]
```diff
@@ -219,27 +223,326 @@ foam.CLASS({
...
+            );
+          },
+          errorMessage: 'INVALID_POSTAL_CODE_ERR_MSG'
+        },
+        // Austria
+        {
+          args: ['postalCode', 'countryId'],
+          predicateFactory: function(e) {
+            return e.OR(
+              e.NEQ(foam.nanos.auth.Address.COUNTRY_ID, 'AT'),
```
Comment (jlhughes) [inline src/foam/nanos/auth/Address.js:236]: @kgrgreer can we use mmethod here for switching on country code to get the appropriate reg-ex?

File: src/foam/nanos/auth/Address.js [hunk 2]
```diff
@@ -219,27 +223,326 @@ foam.CLASS({
...
                 foam.nanos.auth.Address.POSTAL_CODE,
-                /^^\d{5}(?:[-\s]\d{4})?$/i)
+                /^\d{5}(?:[-\s]\d{4})?$/i)
+            );
+          },
+          errorMessage: 'INVALID_POSTAL_CODE_ERR_MSG'
+        },
+        // Austria
+        {
+          args: ['postalCode', 'countryId'],
```
Comment (jlhughes) [inline src/foam/nanos/auth/Address.js:233]: validationPredicates - on each call we are walking them all?

File: src/foam/nanos/auth/Address.js (hunk 1 above)
Comment (kgrgreer) [reply src/foam/nanos/auth/Address.js:236]: No, mmethod only works with types. We could make a switch/case mlang to make this easier, but we don't current have such a thing.

File: src/foam/nanos/auth/Address.js (hunk 2 above)
Comment (jinnnn9) [reply src/foam/nanos/auth/Address.js:233]: @jlhughes now you mention it, it doesn't look very efficient. I will make changes to it if we can go with your suggestion above.

**Comment by jlhughes** [comment]
Merging as is. Will track improving validation check logic in another PR.

**Pull request merged**